File: breadth_first_search/breadth_first_search.java
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

public class Node {
    private int name;
    private List<Node> children;

    public Node(int name) {
        this.name = name;
        this.children = new ArrayList<>();
    }

    public Node addChild(int name) {
        this.children.add(new Node(name));
        return this;
    }

    public List<Integer> breadthFirstSearch(List<Integer> array) {
        Queue<Node> queue = new LinkedList<>();
        queue.add(this);
        while (!queue.isEmpty()) {
            Node currentNode = queue.poll();
            array.add(currentNode.name);
            for (Node child : currentNode.children) {
                queue.add(child);
            }
        }
        return array;
    }

    public static void main(String[] args) {
        Node myTree = new Node(1).addChild(2).addChild(3).addChild(4);
        List<Integer> result = myTree.breadthFirstSearch(new ArrayList<>());
        System.out.println(result);  // Output: [1, 2, 3, 4]
    }
}

File: breadth_first_search/breadth_first_search.py
class Node:
  def __init__(self, name) -> None:
      self.children = []
      self.name = name

  def add_child(self, name):
    self.children.append(Node(name))
    return self
  
  def breadth_first_search(self, array):
    queue = [self]
    while len(queue) > 0:
      current_node = queue.pop(0)
      array.append(current_node.name)
      for node in current_node.children:
        queue.append(node)
    return array

myTree = Node(1).add_child(2).add_child(3).add_child(4)
print(myTree.breadth_first_search([]))